cmd/dogstatsd/main.go

```diff
@@ -17,6 +17,7 @@ import (
 	log "github.com/cihub/seelog"
 	"github.com/spf13/cobra"
 
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
```

Comment:
I'm not a fan of introducing coupling between the `cmd`s. Could the relevant logic be moved to `pkg/config` or a `pkg/util/...` one?

Reply:
Sure, I'm happy to do that

Reply:
There is a lot of coupling, this will be a more complex job that I initially thought. We might wanna move most of the common folder (or all of it) to the utils or somewhere

Reply:
That's what I'm afraid of: by importing `github.com/DataDog/datadog-agent/cmd/agent/common` in dsd we could bring in the collector and the corechecks.

Although [common.SetupConfig](https://github.com/DataDog/datadog-agent/blob/29adfc33345d422af77c483cfa10fd99272d89ec/cmd/agent/common/helpers.go#L15) looks like it could be moved to `pkg/config` under a more explicit name by passing it a list of configpaths, instead of having it query them.

Reply:
hmm that's a better idea!

Reply:
the problem is the DistPath

Reply:
It's what's hard to decouple. I started doing what you suggested and ran into that as an issue